src/main/java/org/linksharing/server/auth/AuthController.java

```diff
@@ -47,7 +51,12 @@ public String registerUser(@Valid User user, BindingResult result) {
             return "redirect:/login?userExists";
         }
 
+        UserProfileDetails userProfile = new UserProfileDetails();
+        userProfile.setEmail(user.getEmail());
+        userProfile.setFirstName(user.getUsername());
+
         userRepository.save(user);
+        profileRepository.save(userProfile);
```

Comment:
Don't have to do that here. Move this to `getProfileDetails`.

Reply:
Why shouldn't we? It seems logical to me. When new user registers, his profile creates and goes to DB at the same time. 
No additional checks needed later during CRUD operations. If we get User by Id, we 100% sure that he already has Profile.

Reply:
This doesn't look to me as necessary thing to do but sure, let's keep it that way.